File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        n = len(nums)
        def f(i, last):
            if i == n:
                return 0
            skip = f(i+1, last)
            take = 0
            if last == -1 or nums[i] > nums[last]:
                take = 1 + f(i+1, i)
            return max(skip, take)

        return f(0, -1)

        

File: Data Structures & Algorithms/coin-change/submission-0.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        int n = coins.size();
        vector<vector<int>> dp (n, vector<int>(amount + 1, 0))
        
        for(int i = 1; i < n; i++) {
            for(int j = 0; j <= amount; j++) {
                int skip = dp[i-1][j];
                int take = INT_MAX;
                if(j >= nums[i]) {
                    take = dp[i][j - nums[i]];
                }
            }
        }

        return dp[n-1][amount];
    }
};
